puppet/modules/quickstack/manifests/load_balancer/horizon.pp

```diff
@@ -1,29 +1,69 @@
 class quickstack::load_balancer::horizon (
-  $frontend_pub_host,
-  $frontend_priv_host,
-  $frontend_admin_host,
-  $backend_server_names,
-  $backend_server_addrs,
-  $port = '80',
-  $mode = 'http',
-  $log = 'httplog',
+ $frontend_pub_host,
+ $frontend_priv_host,
+ $frontend_admin_host,
+ $backend_server_names,
+ $backend_server_addrs,
+ $backend_port = '80',
+ $mode = 'http',
+ $option = 'httplog'
 ) {
 
-  include quickstack::load_balancer::common
+ include quickstack::load_balancer::common
 
-  quickstack::load_balancer::proxy { 'horizon':
+  if str2bool("$::haproxy_cert_exist") {
+    quickstack::load_balancer::proxy { 'horizon':
```

Comment:
why not refactor this and the repeated declaration below the if so it is only described once?  port could easily stay a variable, since that is different between the ssl/non-ssl versions, and you could just set backend_port to nil/undef (unless this causes haproxy config to be incorrect, but passing undef as the value should be the same as not even calling it, as I understand it).  Then you can just use this if block to set those params and perhaps create the horizon_http resource.